orttraining/orttraining/python/training/ortmodule/debug_options.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# debug_options.py
+
+import os
+
+from ._logger import LogLevel
+
+class _SaveOnnxOptions:
+    """Configurable option to save ORTModule intermediate onnx models."""
+
+    # class variable
+    _path_environment_key = 'ORTMODULE_SAVE_ONNX_PATH'
+
+    def __init__(self, save, name_prefix):
+        self._save, self._name_prefix, self._path = self._extract_info(save, name_prefix)
+
+    def _extract_info(self, save, name_prefix):
+        # get the destination path from os env variable
+        destination_path = os.getenv(_SaveOnnxOptions._path_environment_key, os.getcwd())
+        # perform validation only when save is True
+        if save:
+            self._validate(save, name_prefix, destination_path)
+        return save, name_prefix, destination_path
+
+    def _validate(self, save, name_prefix, destination_path):
+        # check if directory is writable
+        if not os.access(destination_path, os.W_OK):
+            raise OSError(f"Directory {destination_path} is not writable. Please set the {_SaveOnnxOptions._path_environment_key} environment variable to a writable path.")
+
+        # check if input prefix is a string
+        if not isinstance(name_prefix, str):
+            raise TypeError(f"Expected name prefix of type str, got {type(name_prefix)}.")
+
+        # if save_onnx is set, save_onnx_prefix must be a non empty string
+        if not name_prefix:
```

Comment:
Not blocking for the PR, but maybe candidate for follow up PRs, I would perform more validation here, such as the presence of '/' or '\' (users passing path to the prefix) or any other funny char that might break due to differences in windows/linux/mac support or maximum length

Reply:
This might be a good idea. But it puts a heavy onus on us to check for these traits in the ```name_prefix``` (parsing strings for chars, etc). I think we could let it fail and let the user correct their ```name_prefix``` from the error they receive if they so desire?

Reply:
I can address this in a subsequent pull request if we desire to add this check.

Reply:
Agreed, it makes sense